Remove activesupport dependency

sferik opened the pull request:
This library is only using Active Support in two places, so I thought it would be better to remove it:
1. Symbolization of keys
   - This is implemented as an extension of `Hash`.
2. Calculating the length of multibyte strings.
   - This can be done using String#unpack("U*")

I've also include a few other clean-up commits, which you can cherry-pick around if you prefer.

`ActiveSupport::JSON` was also being used in the tests but that can be replaced with `JSON`, which is part of the standard library as of Ruby 1.9. I added the `json` gem as a development dependency for the off chance someone is doing development in Ruby 1.8.

Tests are passing on MRI 1.9.3-p194.

sferik committed:
Remove activesupport dependency

Add unf dependency for doing Unicode normalization.

keitaf commented:
I made another pull request https://github.com/twitter/twitter-text-rb/pull/68 to make HitHighlight work correctly w/o ActiveSupport. Can you please review it?

keitaf commented on lib/twitter-text/validation.rb:
```diff
@@ -22,7 +22,7 @@ module Validation extend self
       options = DEFAULT_TCO_URL_LENGTHS.merge(options)
 
-      length = ActiveSupport::Multibyte::Chars.new(text).normalize(:c).length
+      length = text.unpack("U*").length
```
I guess this doesn't perform Unicode Normalization Form NCF. Is there any unicode gem we can use instead?

sferik commented:
I've rebased from `HEAD` and added a dependency on the `unf` gem to handle Unicode Normalization Form.

Now, this patch might not seem so useful (trading one dependency for another) but I believe that the `unf` dependency is (much) more precise than depending on `activesupport`.

niw commented:
-1 

Oops, sorry for the last comment, I removed it. unf.gem for MRI is in the separated gem. but still -1.
Because unf.gem is using C code, this might have some troubles in some cases.

sferik commented:
The [`unf`](https://rubygems.org/gems/unf) gem was designed to work with JRuby as well as CRuby. There's a [separate version of the gem](https://rubygems.org/gems/unf/versions) built for the JRuby platform, which does not use any C code. I have not tested other implementations like Rubinius or MagLev but they both aim to support C extensions, so I don't see why this is an issue.

sferik commented:
@jakl I just rebased again so this patch applies cleanly to `master`. Travis is reporting some failures in the conformance tests but, as far as I can tell, these same failures are occurring in `master`.
